hibernate-core/src/main/java/org/hibernate/engine/jdbc/spi/SqlStatementLogger.java

```diff
@@ -105,4 +119,30 @@ public void logStatement(String statement, Formatter formatter) {
 			System.out.println( "Hibernate: " + statement );
 		}
 	}
+
+	/**
+	 * Log a slow SQL query
+	 *
+	 * @param sql The SQL query.
+	 * @param startTime Start time in milliseconds.
+	 */
+	@AllowSysOut
+	public void logSlowQuery(String sql, long startTime) {
+		if ( logSlowQuery < 1 ) {
+			return;
+		}
+		assert startTime > 0 : "startTime is invalid!";
```

Comment:
wouldn't all start times be invalid then?

Reply:
if the feature is enabled, startTime must have a valid value.